File: _book/01_front_end_fundamentals/d04/David_Hayes/reps.js
function lengths(arrayOfStrings) {
  var arrayOfLengths = [];
  for (var str in arrayOfStrings) {
    arrayOfLengths[str] = arrayOfStrings[str].length;
  }
  return arrayOfLengths;
}

function transmogrifier(first, second, power) {
  return Math.pow(first * second,power);
}

function toonify(accent,sentence) {
  if (accent=="daffy") {
    sentence = sentence.replace(/[Ss]/g, 'th');
  } else if (accent=="elmer") {
    sentence = sentence.replace(/[Rr]/g, 'w');
  }
  return sentence;
}

function wordReverse(str) {
  var orgString = str.split(' ');
  var revString = orgString.reverse();
  var outString = "";
  for (var wrd in revString) {
    if (wrd==0) {
      outString = outString + revString[wrd];
    } else {
      outString = outString + " " + revString[wrd];
    }
  }
  return outString;
}

function letterReverse(str) {
  var nextwrd;
  var orgString = str.split(' ');
  var wrdArray = [];
  var outString = "";
  for (var wrd in orgString) {
    nextwrd = orgString[wrd];
    wrdArray = nextwrd.split("");
    revArray = wrdArray.reverse();
    if (wrd != 0) {
      outString = outString + " "
    }
    for (var ltr in revArray) {
      outString =  outString + revArray[ltr];
    }
  }
  return outString;
}

function longest(wordArray) {
var lword = "";
var myword;
  for (var word in wordArray) {
   myword = wordArray[word];
   if (wordArray[word].length > lword.length) {
      lword = wordArray[word];
    }
  }
  return lword;
}

function repMaster(inputString,stringFn) {
  return stringFn(inputString);
}

File: 04_ruby/d01/Matt_Herron/reps.rb
## Round 1
def lengths(arr)
  arr.map { |s| s.length }
end

## Round 2
def transmogrifier(a, b, c)
  (a * b) ** c
end

## Round 3
def toonify(accent, sentence)
  if accent.downcase == 'daffy'
    return sentence.gsub 's', 'th'
  elsif accent.downcase == 'elmer'
    return sentence.gsub 'r', 'w'
  else
    return sentence
  end
end

## Round 4
def wordReverse(s)
  s.split(' ').reverse.join(' ')
end

## Round 5
def letterReverse(sentence)
  sentence.split(' ').map { |s| s.split('').reverse.join('') }.join(' ')
end

## Round 6
def longest(arrs)
  lengths = arrs.map{ |s| s.length }
  arrs[lengths.rindex(lengths.max)]
end
